packages/x-charts/src/internals/plugins/featurePlugins/useChartKeyboardNavigation/useChartKeyboardNavigation.ts

```diff
@@ -0,0 +1,278 @@
+            seriesId,
+            dataIndex:
+              (dataLength + (state.keyboardNavigation.item?.dataIndex ?? 1) - 1) % dataLength,
+          },
+        },
+      };
+    });
+  });
+
+  const focusPreviousSeries = useEventCallback(function focusPreviousSeries() {
+    let setNewSeries = false;
+    store.update((state) => {
+      let { type, seriesId } = state.keyboardNavigation.item ?? {};
+
+      const previousSeries = getPreviousSeriesWithData(
+        state.series.processedSeries,
+        type,
+        seriesId,
+      );
+      if (previousSeries === null) {
+        return {
+          ...state,
+          keyboardNavigation: {
+            ...state.keyboardNavigation,
+            item: null, // No series to move the focus too.
+          },
+        };
+      }
+      type = previousSeries.type;
+      seriesId = previousSeries.seriesId;
+
+      const dataLength = state.series.processedSeries[type]!.series[seriesId].data.length;
+
+      setNewSeries = true;
+      return {
+        ...state,
+        keyboardNavigation: {
+          ...state.keyboardNavigation,
+          item: {
+            type,
+            seriesId,
+            dataIndex: Math.min(dataLength - 1, state.keyboardNavigation.item?.dataIndex ?? 0),
+          },
+        },
+      };
+    });
+    return setNewSeries;
+  });
+
+  const focusNextSeries = useEventCallback(function focusNextSeries() {
+    let setNewSeries = false;
+
+    store.update((state) => {
+      let { type, seriesId } = state.keyboardNavigation.item ?? {};
+
+      const nextSeries = getNextSeriesWithData(state.series.processedSeries, type, seriesId);
+
+      if (nextSeries === null) {
+        return {
+          ...state,
+          keyboardNavigation: {
+            ...state.keyboardNavigation,
+            item: null, // No series to move the focus too.
+          },
+        };
+      }
+      type = nextSeries.type;
+      seriesId = nextSeries.seriesId;
+
+      const dataLength = state.series.processedSeries[type]!.series[seriesId].data.length;
+
+      setNewSeries = true;
+      return {
+        ...state,
+        keyboardNavigation: {
+          ...state.keyboardNavigation,
+          item: {
+            type,
+            seriesId,
+            dataIndex: Math.min(dataLength - 1, state.keyboardNavigation.item?.dataIndex ?? 0),
+          },
+        },
+      };
+    });
+
+    return setNewSeries;
+  });
+
+  const removeFocus = useEventCallback(function removeFocus() {
+    store.update((state) => {
+      if (state.keyboardNavigation.item === null) {
+        return state;
+      }
+      return {
+        ...state,
+        keyboardNavigation: {
+          ...state.keyboardNavigation,
+          item: null,
+        },
+      };
+    });
+  });
+
+  React.useEffect(() => {
+    const element = svgRef.current;
+    if (!element || !params.enableKeyboardNavigation) {
+      return undefined;
+    }
+
+    function keyBoardHandler(event: KeyboardEvent) {
```

Comment:
Keyboard is a single word

```suggestion
    function keyboardHandler(event: KeyboardEvent) {
```